### PR TITLE
Optimizing google play implementation

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

Modify the interface to execute in the cocos thread, since google's interface is labeled AnyThread

image

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: @types/jsb.d.ts [hunk 1]
```diff
@@ -1411,9 +1411,9 @@ declare namespace jsb {
         isExternalOfferAvailableAsync: (listener: ExternalOfferAvailabilityListener) => void;
 
         isFeatureSupported: (productType: string) => BillingResult;
-        showAlternativeBillingOnlyInformationDialog: (listener: AlternativeBillingOnlyInformationDialogListener) => BillingResult;
```
Comment (qiuguohua) [inline @types/jsb.d.ts:1414]: These interfaces are required to be executed on the UI thread, so it is not possible to get the return value

File: native/vendor/google/billing/GoogleBilling.h [hunk 2]
```diff
@@ -90,8 +90,8 @@ class CC_DLL BillingClient : public cc::RefCounted {
     private:
         friend class BillingClient;
         friend class JniBilling;
-        bool _enableAlternativeBillingOnly;
-        bool _enableExternalOffer;
+        bool _enableAlternativeBillingOnly{false};
+        bool _enableExternalOffer{false};
         PendingPurchasesParams* _pendingPurchasesParams;
         scopedListener _purchasesUpdatedListener;
         scopedListener _userChoiceBillingListener;
```
Comment (dumganhar) [inline native/vendor/google/billing/GoogleBilling.h:97]: Need to initialize all variables.

File: native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java [hunk 3]
```diff
@@ -56,36 +56,27 @@ of this software and associated documentation files (the "Software"), to deal
 import com.android.billingclient.api.UserChoiceDetails;
 import com.cocos.lib.CocosHelper;
 import com.cocos.lib.GlobalObject;
+
+import android.nfc.Tag;
```
Comment (dumganhar) [inline native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java]: Don't import `android.nfc.Tag;`

File: native/vendor/google/billing/GoogleBilling.h (hunk 2 above)
Comment (qiuguohua) [reply native/vendor/google/billing/GoogleBilling.h:97]: Fixed

File: native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java (hunk 3 above)
Comment (qiuguohua) [reply native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java]: Fixed

File: native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java [hunk 4]
```diff
@@ -256,280 +236,263 @@ public static boolean isReady(int tag) {
...
+        if (billing == null) {
+            return;
         }
-        return BillingResult.newBuilder().build();
+        GlobalObject.runOnUiThread(new Runnable() {
+           @Override
+           public void run() {
+               BillingResult res = billing.showExternalOfferInformationDialog(new ExternalOfferInformationDialogListener() {
+                   @Override
+                   public void onExternalOfferInformationDialogResponse(@NonNull BillingResult billingResult) {
```
Comment (dumganhar) [inline native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java:458]: Are `res` and `billingResult` the same object ?

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Comment by qiuguohua** [comment]
https://github.com/cocos/3d-tasks/issues/18696

**Pull request merged**